test/e2e/logs_test.go

```diff
@@ -0,0 +1,47 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// testNodeLogs ensures that all required log files were created, and copies them to the test's artifact directory
+func testNodeLogs(t *testing.T) {
+	// All these paths are relative to /var/log/
+	logFiles := []string{
+		"kube-proxy\\kube-proxy.exe.INFO",
+		"kube-proxy\\kube-proxy.exe.ERROR",
+		"kube-proxy\\kube-proxy.exe.WARNING",
+		"hybrid-overlay\\hybridOverlay.log",
+		"kubelet\\kubelet.log",
+	}
+
+	nodeArtifacts := filepath.Join(os.Getenv("ARTIFACT_DIR"), "nodes")
+	for _, node := range gc.nodes {
+		nodeDir := filepath.Join(nodeArtifacts, node.Name)
+		for _, file := range logFiles {
+			// Use oc to read the expected log files
+			cmd := exec.Command("oc", "adm", "node-logs", "--path", file, node.Name)
+			out, err := cmd.Output()
+			require.NoError(t, err)
+
+			// Save log files to the artifact directory
+			splitPath := strings.Split(file, "\\")
```

Comment:
You're saving it logfiles to `ARTIFACT_DIR` as part of this code? That should have been log collection in my opinion.